pkg/controller/nodelifecycle/node_lifecycle_controller.go

```diff
@@ -1040,8 +1040,8 @@ func (nc *Controller) tryUpdateNodeHealth(ctx context.Context, node *v1.Node) (t
 		} else {
 			transitionTime = nodeHealth.readyTransitionTimestamp
 		}
-		if klog.V(5).Enabled() {
-			klog.Infof("Node %s ReadyCondition updated. Updating timestamp: %+v vs %+v.", node.Name, nodeHealth.status, node.Status)
+		if klogV := klog.V(5); klogV.Enabled() {
+			klogV.Infof("Node %s ReadyCondition updated. Updating timestamp: %+v vs %+v.", node.Name, nodeHealth.status, node.Status)
```

Comment:
I am not sure why the original code checked with `klog.V(5).Enabled()` beforehand. The parameters seem to be fairly simple.

I'd probably convert this to a single `klog.V(5).Infof` call but first wanted to use this example to ask when this code pattern with an explicit prior verbosity check makes sense?

Reply:
`Enabled()` function checks whether the log level is enabled or not see https://github.com/kubernetes/klog/blob/main/klog.go#L1447-L1452
We also check something similar when we use something like `klog.V(5).Infof` see https://github.com/kubernetes/klog/blob/main/klog.go#L1417-L1420
Not sure about the purpose of `klog.V(5).Enabled()` in the original code, but it makes sense to me if we remove this now.

Reply:
It make sense to check in advance if preparing the parameters is expensive. We agree that this doesn't seem to be the case here, so I'll remove it.

Reply:
Looking again I noticed that `node.Status` is large. The intention probably was to log that additional data only at a higher log level. Let's keep that.